src/backend/utils/adt/like.c

```diff
@@ -161,6 +164,11 @@ GenericMatchText(const char *s, int slen, const char *p, int plen, Oid collation
 					 errmsg("nondeterministic collations are not supported for LIKE")));
 	}
 
+	if (sql_dialect == SQL_DIALECT_TSQL && set_like_collation_hook)
```

Comment:
From my understanding, TSQL LIKE corresponds to PG ILIKE. In which case, do we need to call this (`GenericMatchText`) function?

Reply:
When compare with deterministic collations, and also some loophole like BABEL-4046, it'll go into GenericMatchText

If it goes into GenericMatchText, we don't want the like execution got crashed with cid not set

that's why also set cid in GenericMatchText